responsibleai/responsibleai/_managers/counterfactual_manager.py

```diff
@@ -119,6 +228,90 @@ def get_result(self):
+        if cf_result[CounterfactualConfig.COUNTERFACTUAL_OBJ] is not None:
+            counterfactuals_dict = json.loads(
+                cf_result[CounterfactualConfig.COUNTERFACTUAL_OBJ]
+            )
+
+            schema_version = _get_schema_version(counterfactuals_dict)
+            if schema_version == _SchemaVersions.V1:
+                cf_schema_keys = _V1SchemaConstants.ALL
+            else:
+                cf_schema_keys = _V2SchemaConstants.ALL
+
+            for counterfactual_examples_key in cf_schema_keys:
+                file_path = (data_directory_path /
+                             (counterfactual_examples_key + '.json'))
+                with open(file_path, 'w') as file_path:
+                    json.dump(
+                        counterfactuals_dict[counterfactual_examples_key],
+                        file_path)
+
+        file_path = (data_directory_path /
+                     (CounterfactualConfig.HAS_COMPUTATION_FAILED + '.json'))
+        with open(file_path, 'w') as file_path:
+            json.dump(
+                cf_result[CounterfactualConfig.HAS_COMPUTATION_FAILED],
+                file_path)
+
+        file_path = (data_directory_path /
+                     (CounterfactualConfig.FAILURE_REASON + '.json'))
+        with open(file_path, 'w') as file_path:
+            json.dump(
+                cf_result[CounterfactualConfig.FAILURE_REASON],
+                file_path)
+
+        file_path = (data_directory_path /
+                     (CounterfactualConfig.IS_COMPUTED + '.json'))
+        with open(file_path, 'w') as file_path:
+            json.dump(
+                cf_result[CounterfactualConfig.IS_COMPUTED],
+                file_path)
+
+    def load_result(self, data_directory_path):
+        metadata_file_path = (data_directory_path /
+                              (_CommonSchemaConstants.METADATA + '.json'))
+
+        if metadata_file_path.exists():
+            with open(metadata_file_path, 'r') as result_file:
+                metadata = json.load(result_file)
+
+            if metadata['version'] == _SchemaVersions.V1:
+                cf_schema_keys = _V1SchemaConstants.ALL
+            else:
+                cf_schema_keys = _V2SchemaConstants.ALL
+
+            counterfactual_examples_dict = {}
+            for counterfactual_examples_key in cf_schema_keys:
+                result_path = (data_directory_path /
+                               (counterfactual_examples_key + '.json'))
+                with open(result_path, 'r') as result_file:
+                    counterfactual_examples_dict[
+                        counterfactual_examples_key] = json.load(result_file)
+
+            counterfactuals_json_str = json.dumps(counterfactual_examples_dict)
+            self.counterfactual_obj = \
+                CounterfactualExplanations.from_json(counterfactuals_json_str)
+        else:
+            self.counterfactual_obj = None
+
+        result_path = (data_directory_path /
```

Comment:
I think this kind of logic needs to live in a common place instead of being replicated in each manager?
HAS_COMPUTATION_FAILED
FAILURE_REASON
IS_COMPUTED
these are all common to all of the managers and seems strange to have this logic just here?

Reply:
Yes, I agree that we should consolidate the logic as per common code. But the major trouble is that causal is using a completely different logic to manage the config. So we might need to refactor the causal manager first before addressing the consolidation of serilization logic.